include/HamiltonianParameters/AtomicDecompositionParameters.hpp

```diff
@@ -0,0 +1,165 @@
+// This file is part of GQCG-gqcp.
+//
+// Copyright (C) 2017-2019  the GQCG developers
+//
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+//
+#ifndef GQCP_ATOMICDECOMPOSITIONPARAMETERS_HPP
+#define GQCP_ATOMICDECOMPOSITIONPARAMETERS_HPP
+
+
+#include <Molecule.hpp>
+#include "HamiltonianParameters/HamiltonianParameters.hpp"
+
+namespace GQCP {
+
+
+/**
+ *  A struct that holds the collection of Hamiltonian Parameters that represent different molecular decompositions.
+ */
+struct AtomicDecompositionParameters {
```

Comment:
This is a fairly elaborate struct: maybe make this into a class and put the source code into a `.cpp`-file?